runtime/android/core_internal/src/org/xwalk/core/internal/XWalkDownloadListenerImplInternal.java

```diff
@@ -23,10 +23,10 @@
 import java.io.OutputStream;
 
 import org.xwalk.core.internal.AndroidProtocolHandler;
-import org.xwalk.core.internal.DownloadListener;
 import org.xwalk.core.internal.R;
+import org.xwalk.core.internal.XWalkDownloadListenerInternal;
 
-class XWalkDownloadListenerImpl implements DownloadListener {
+class XWalkDownloadListenerImplInternal extends XWalkDownloadListenerInternal {
```

Comment:
It's okay to keep the name `XWalkDownloadListenerImpl`. If a class name is postfixed with `Internal`, it generally means we intend to expose it and use it as embedding API